include/seahorn/Expr/TypeChecker.hh

```diff
@@ -1,17 +1,168 @@
 #pragma once
-
-#include "seahorn/Expr/ExprApi.hh"
+#include "seahorn/Expr/ExprCore.hh"
+#include "seahorn/Expr/ExprOpSort.hh"
 
 namespace expr {
 class TypeChecker {
   class Impl;
-  Impl* m_impl;
+  Impl *m_impl;
 
 public:
   TypeChecker();
   ~TypeChecker();
   Expr sortOf(Expr e) { return this->typeOf(e); }
-  Expr typeOf(Expr e); 
+  Expr typeOf(Expr e);
```

Comment:
comment each method. These are user facing and need substantive documentation.
substantive == 3-4 lines